### PR TITLE
fix(#914): remove existence check before create-file, rename-file, harden clobbered buffer renaming

### LINKED ISSUE #914: Case sensitive rename not possible (cannot rename)
**OS**
macOS Monterey 12.1

**Neovim version**
v0.6.1

**Nvim-tree version**
`master`

**Describe the bug**
Renaming a file like `Something.md` to `something.md` does not work. **[NvimTree] Cannot rename: file already exists** If I rename it first to something different, it works: `Something.md` → `something2.md`  → `something.md` .

**Expected behavior**
Renaming a file respecting lower and upper case is possible. `Something.md` → `something.md`.

Thanks for your work 🙏🏻

### PR DESCRIPTION
**Pull request by alex-courtis**
fixes #914

### CODE REVIEW

**Comment by gegoune** [comment]
I hope that it will not bite back. :)

**Review by gegoune — APPROVED** [review]

**Comment by alex-courtis** [comment]
Rename failure case OK:
`[NvimTree] "Cannot rename /home/alex/914/File.txt -> /home/alex/914/inexistent/File.txt: ENOENT: no such file or directory: /home/alex/914/File.txt -> /home/alex/914/inexistent/File.txt"`

Rename when both buffers open failed:
```
E5108: Error executing lua: ...t-dev/site/pack/packer/start/dev/lua/nvim-tree/utils.lua:166: Failed to rename buffer
stack traceback:
        [C]: in function 'nvim_buf_set_name'
        ...t-dev/site/pack/packer/start/dev/lua/nvim-tree/utils.lua:166: in function 'rename_loaded_buffers'
        ...acker/start/dev/lua/nvim-tree/actions/fs/rename-file.lua:25: in function 'rename'
        ...acker/start/dev/lua/nvim-tree/actions/fs/rename-file.lua:75: in function 'on_confirm'
        /usr/share/nvim/runtime/lua/vim/ui.lua:93: in function 'input'
        ...acker/start/dev/lua/nvim-tree/actions/fs/rename-file.lua:69: in function 'handle_tree_actions'
        ...pack/packer/start/dev/lua/nvim-tree/actions/dispatch.lua:123: in function 'dispatch'
        ...ite/pack/packer/start/dev/lua/nvim-tree/actions/init.lua:273: in function <...ite/pack/packer/start/dev/lua/nvim-tree/actions/init.lua:272>
```

Create clobbers OK, although vim will need to reload the external changes if the buffer is open.

File: lua/nvim-tree/utils.lua [hunk 1]
```diff
@@ -154,27 +154,35 @@ function M.get_nodes_by_line(nodes_all, line_start)
...
+function M.rename_loaded_buffers(from_path, to_path)
+  -- find from then to
+  local from_buf, to_buf
   for _, buf in pairs(vim.api.nvim_list_bufs()) do
     if vim.api.nvim_buf_is_loaded(buf) then
-      local buf_name = vim.api.nvim_buf_get_name(buf)
-      local exact_match = buf_name == old_path
-      local child_match = (
-        buf_name:sub(1, #old_path) == old_path and buf_name:sub(#old_path + 1, #old_path + 1) == path_separator
-      )
```
Comment (alex-courtis) [inline lua/nvim-tree/utils.lua:164]: All callers use absolute paths hence this fuzzy matching can be removed.

**Comment by alex-courtis** [comment]
I was tempted to remove the existence checks from paste, however they can be useful and nobody has reported issues.

**Comment by alex-courtis** [comment]
This was a failure; trying a simpler approach: #1988

**Pull request closed without merging**